### DISCUSSION TITLE
Add X-Robots-Tag to all hosts.

### DISCUSSION THREAD

**Discussion by cowpiebrownrice** [NginxProxyManager/nginx-proxy-manager#3978] [Q&A]
I installed a Nginx Reverse Proxy via docker.
I would like to add `X-Robots-Tag` to all hosts. I tried adding the option:
```
add_header X-Robots-Tag "noindex, nofollow, nosnippet, noarchive";
```
in `Advanced / Custom Nginx Configuration`
but it doesn’t work.
I read that the `add_header` options if not included in a custom location do not work.

What custom location can I use? Or does anyone know another way to make it default?

**Comment by rodrigolyamamoto** [comment]
I had a similar issue, but it was related to security headers.

I added my security headers under **Proxy Hosts → Edit → Settings → Custom Nginx Configuration**. These directives are added to the `server { }` block.

However, the `location / { }` block has `include conf.d/include/proxy.conf,` which contains:

add_header X-Served-By $host;

This overrides all `add_header` directives defined in the server block.

To fix it, I created a **Custom Location** with the path `/` and configured the same Scheme, Forward Hostname/IP, and Forward Port as in the Details tab. After saving the configuration, everything worked as expected.

**Comment by gitsult4n** [comment]
There is a global hook for this
Every proxy host server block ends with

```nginx
include /data/nginx/custom/server_proxy[.]conf;
```

So create `/data/nginx/custom/server_proxy.conf` once

```nginx
add_header X-Robots-Tag "noindex, nofollow, noarchive" always;
```

Then restart the container and every host picks it up with no per host edits

Other hooks in the same folder if you need wider scope

- `http.conf` and `http_top.conf` for the whole http context
- `root_top.conf` for the main context
- `server_dead.conf` and `server_redirect.conf` for those host types

Keep `always` on the header or it is dropped on error responses
And remember `add_header` in a location wipes inherited ones so any custom location needs it repeated
